### ISSUE TITLE
Remote arguments validations not supported for non-scalar types

### ISSUE THREAD

**Issue by tirumaraiselvan** [tirumaraiselvan/graphql-engine#9]
Remote arguments are able to validate scalar fields but not complex fields like List, Object etc

E.g.

```
type: create_remote_relationship
args:
  name: message
  table: profiles
  hasura_fields:
    - id
  remote_schema: my-remote-schema
  remote_field: messages
  remote_arguments:
    where:
      id:
        _eq: "$id"
```

**Comment by tirumaraiselvan** [comment]
Maybe the way to solve it is to re-use/take inspiration from `GraphQL.Validate.InputValue` module.

**Comment by chrisdone** [comment]
Copying my comment over from PR 1:

> I think that if I extend the python fake graphql service to expect argument format like that, then it'll be easier to test for objects. Will we support lists? I don't see a reason not to, but you might? Will we support enums? 

> For later reference, can you point to an example of how I might do that on the python side? I'm not sure what to do in [the python here](https://github.com/hasura/graphql-engine/blob/500089308832d15bf2f29583b2f227226c0f2526/server/tests-py/graphql_server.py#L611). I'm confident I could figure out the Haskell side.

**Comment by chrisdone** [comment]
Copying @tirumaraiselvan's reply over:

> True. We might need to support everything. For e.g. what if the join query is something like:
```
messages(include: [$id]) {
   id
   name
}
```
>I can extend the python server to have more complicated types.

**Comment by tirumaraiselvan** [comment]
We do not have ENUM type which can be a hasura_field but an ENUM type might be used as a static value in the remote args.

**Comment by chrisdone** [comment]
Right, a static string value which could be passed to the remote server without issue... :thinking:

**Comment by tirumaraiselvan** [comment]
Closed by #11
